src/devices/Uln2003/Uln2003.cs

```diff
@@ -0,0 +1,169 @@
+                 { true, false, false, false, true, false, false, false },
+                 { false, true, false, false, false, true, false, false },
+                 { false, false, true, false, false, false, true, false },
+                 { false, false, false, true, false, false, false, true }
+            };
+
+        static bool[,] _fullStepDualPhaseSequence = new bool[4, 8] {
+                 { true, false, false, true, true, false, false, true },
+                 { true, true, false, false, true, true, false, false },
+                 { false, true, true, false, false, true, true, false },
+                 { false, false, true, true, false, false, true, true }
+            };
+
+        /// <summary>
+        /// Initialize a Uln2003 class.
+        /// </summary>
+        /// <param name="pin1">The GPIO pin number which corresponds pin A on ULN2003 driver board.</param>
+        /// <param name="pin2">The GPIO pin number which corresponds pin B on ULN2003 driver board.</param>
+        /// <param name="pin3">The GPIO pin number which corresponds pin C on ULN2003 driver board.</param>
+        /// <param name="pin4">The GPIO pin number which corresponds pin D on ULN2003 driver board.</param>
+        public Uln2003(int pin1, int pin2, int pin3, int pin4)
+        {
+            _pin1 = pin1;
+            _pin2 = pin2;
+            _pin3 = pin3;
+            _pin4 = pin4;
+
+            _controller = new GpioController();
+
+            _controller.OpenPin(_pin1, PinMode.Output);
+            _controller.OpenPin(_pin2, PinMode.Output);
+            _controller.OpenPin(_pin3, PinMode.Output);
+            _controller.OpenPin(_pin4, PinMode.Output);
+        }
+
+        /// <summary>
+        /// Sets the motor speed to revolutions per minute.
+        /// </summary>
+        /// <remarks>Default revolutions per minute for 28BYJ-48 is approximately 15.</remarks>
+        public short RPM { get; set; }
+
+        /// <summary>
+        /// Sets the stepper's mode.
+        /// </summary>
+        public StepperMode Mode
+        {
+            get
+            {
+                return _mode;
+            }
+            set
+            {
+                _mode = value;
+
+                switch (_mode)
+                {
+                    case StepperMode.HalfStep:
+                        _currentSwitchingSequence = _halfStepSequence;
+                        break;
+                    case StepperMode.FullStepSinglePhase:
+                        _currentSwitchingSequence = _fullStepSinglePhaseSequence;
+                        break;
+                    case StepperMode.FullStepDualPhase:
+                        _currentSwitchingSequence = _fullStepDualPhaseSequence;
+                        break;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Stop the motor.
+        /// </summary>
+        public void Stop()
+        {
+            _steps = 0;
+            _stopwatch.Stop();
```

Comment:
seems to me you could end up in a case where this is either null or hasn't been started. We should definitely make sure that we check for that here.